src/plugins/intel_gpu/src/graph/include/custom_gpu_primitive_inst.h

```diff
@@ -19,12 +20,14 @@ class typed_primitive_inst<custom_gpu_primitive> : public typed_primitive_inst_b
 
 public:
     template<typename ShapeType>
-    static std::vector<layout> calc_output_layouts(custom_gpu_primitive_node const& /*node*/, const kernel_impl_params& impl_param) {
```

Comment:
Can you please also override get_shape_infer_dependencies() function to return an empty vector? This will help avoid runtime synchronization during shape inference, since our custom ops currently don't require the actual data - only the shapes. By default get_shape_infer_dependencies() returns all inputs as dependencies, which forces us to wait for their computation before performing shape inference

Reply:
Added